lib/shoryuken/extensions/active_job_adapter.rb

```diff
@@ -75,6 +75,16 @@ class JobWrapper #:nodoc:
         shoryuken_options body_parser: :json, auto_delete: true
 
         def perform(sqs_msg, hash)
+          approximate_receive_count = sqs_msg.attributes['ApproximateReceiveCount'].to_i
+          if approximate_receive_count > 1
+            message_queue_name = JSON.parse(sqs_msg.body)['queue_name']
+            redrive_policy_json = JSON.parse(Shoryuken::Client.queues(message_queue_name).redrive_policy)
```

Comment:
I'd rather rethink about approach than introduce caching. Caching is nasty stuff.
I think we can infer things from configuration, rather than live system state.